cli/src/test/java/com/devonfw/tools/ide/util/GitUtilsProcessResultMock.java

```diff
@@ -0,0 +1,48 @@
+package com.devonfw.tools.ide.util;
+
+import java.util.List;
+
+import com.devonfw.tools.ide.process.ProcessResult;
+
+/**
+ * Mocks a process result with adjustable params
+ */
+public class GitUtilsProcessResultMock implements ProcessResult {
```

Comment:
It would be smarter to add a `GitHelper` interface and Refactor `GitUtils` to a `GitHelperImpl` implementation of it.
Then you can change this class to a `GitHelperMock` implementing `GitHelper` interface.
This way our test implementation of `IdeContext` could use `GitHelperMock` without loosing the ability to call processes via `ProcessContext`.

Reply:
Adjusted, added new GitContext interface and implementation. This class was removed now.